contracts/TokenSugar.vy

```diff
@@ -89,27 +89,51 @@ def tokens(_limit: uint256, _offset: uint256, _account: address, \
     tokens: address[2] = [staticcall pool.token0(), staticcall pool.token1()]
 
     for i: uint256 in range(2):
-      if tokens[i] in seen:
+      token_index: uint256 = len(seen)
+      seen_count = len(seen)
+
+      # find index of seen token
+      for j: uint256 in range(0, MAX_TOKENS):
+        if j >= seen_count:
+          break
+        if seen[j] == tokens[i]:
+          token_index = j
+          break
+      
```

Comment:
A bit worried we're extensively looping through all tokens all the time to find the index. So I think we need to refactor the whole thing to bring down the complexity.

Probably worth just iterating over all pools first to fetch JUST emerging tokens after what we iterate again, but now knowing which tokens are emerging 🤔